libs/settings/src/setting_collection.cpp

```diff
@@ -103,8 +106,36 @@ void SettingCollection::UpdateFromArgs(int argc, char **argv)
       // update the setting
       std::istringstream iss{cmd_value};
       iss >> *setting;
+      settings_changed.insert(setting->name());
     }
   }
+
+  if (settings_changed.size() != parser.param_size())
+  {
+    FETCH_LOG_ERROR(LOGGING_NAME, "Unrecognised parameter(s) passed.");
+
+    for (auto const &i : parser.params())
+    {
+      auto &parameter_name = i.first;
+      if (settings_changed.find(parameter_name) == settings_changed.end())
+      {
+        FETCH_LOG_ERROR(LOGGING_NAME, "Unrecognised parameter: -", parameter_name);
+      }
+    }
+
+    std::ostringstream oss;
+    oss << "\nValid parameters:";
+
+    for (auto const &setting : settings_)
+    {
+      oss << "\n-" << setting->name() << std::setw(int(30 - setting->name().size())) << " \""
+          << setting->description() << "\"";
+    }
+
+    FETCH_LOG_INFO(LOGGING_NAME, oss.str());
+
+    exit(1);
```

Comment:
To improve testability, `std::exit` (or a wrapper) should be injected so it can be mocked out.